crates/red_knot_python_semantic/resources/mdtest/annotations/any.md

```diff
@@ -0,0 +1,68 @@
+# Any
+
+## Annotation
+
+`typing.Any` is a way to name the Any type.
+
+```py
+from typing import Any
+
+x: Any = 1
+x = "foo"
+
+def f():
+    reveal_type(x)  # revealed: Any
+```
+
+## Aliased to a different name
+
+If you alias `typing.Any` to another name, we still recognize that as a spelling of the Any type.
+
+```py
+from typing import Any as RenamedAny
+
+x: RenamedAny = 1
+x = "foo"
+
+def f():
+    reveal_type(x)  # revealed: Any
+```
+
+## Shadowed class
+
+If you define your own class named `Any`, using that in a type expression refers to your class, and
+isn't a spelling of the Any type.
+
+```py
+class Any:
+    pass
+
+x: Any
+
+def f():
+    reveal_type(x)  # revealed: Any
+
+# This verifies that we're not accidentally seeing typing.Any, since str is assignable
+# to that but not to our locally defined class.
+y: Any = "not an Any"  # error: [invalid-assignment]
+```
+
+## Subclass
+
+The spec allows you to define subclasses of `Any`, which must also resolve to the Any type.
```

Comment:
Sorry, I should have been clearer in my comment about the scope of this PR. I was only intending that we'd return the right `ClassBase` variant when building the MRO, and leave any other updates to the behavior of inheriting-Any as a separate project.

I think the most basic test we can add for this here would just be a test that `Subclass.__mro__` includes `Any/Unknown` when it inherits `Any` explicitly, similar to existing tests in `mro.md`. (That test could live either here or in `mro.md`.)

I've commented below on the semantics that we will want for inheriting-Any, but that doesn't mean I think we should implement those in this PR. Since you've already written some tests, I think it would be reasonable (rather than deleting them) to just update the tests to reflect the current behavior, add TODO comments to reflect the future desired semantics, and leave it like that for now.

If you're interested in implementing more of this behavior as a follow-up PR, that's great! But I'd like to get your first PR landed sooner :)

Reply:
> I was only intending that we'd return the right `ClassBase` variant when building the MRO, and leave any other updates to the behavior of inheriting-Any as a separate project.

Ha, whoops!

> I think the most basic test we can add for this here would just be a test that `Subclass.__mro__` includes `Any/Unknown` when it inherits `Any` explicitly, similar to existing tests in `mro.md`. (That test could live either here or in `mro.md`.)

Done

> I think it would be reasonable (rather than deleting them) to just update the tests to reflect the current behavior, add TODO comments to reflect the future desired semantics, and leave it like that for now.

Done

> I've commented below on the semantics that we will want for inheriting-Any

Ahh, I misunderstood the intented use case in the spec — thought it was introducing a special case where the subclass should be treated exactly like `Any`.  Your explanation makes more sense!